### PR TITLE
Pin jackson2-api temporarily

### PR DESCRIPTION
**Pull request by jetersen**
Pinned jackson2-api to avoid cyclic dependency

We need to pin this to unblock https://github.com/jenkinsci/bom/pull/1011#issuecomment-1100675121

Can be removed once BOM and metrics plugin is updated to using that BOM

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -191,6 +192,13 @@
       <groupId>io.jenkins.configuration-as-code</groupId>
       <artifactId>test-harness</artifactId>
       <scope>test</scope>
+      <!-- TODO https://github.com/jenkinsci/bom/pull/1011 -->
+      <exclusions>
+        <exclusion>
+          <groupId>org.jenkins-ci.plugins</groupId>
+          <artifactId>jackson2-api</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (jetersen) [inline pom.xml:201]: Only temporary until https://github.com/jenkinsci/bom/pull/1011 is resolved.

I could go to test-harness and attempt to make another PR but it would be yet another PR that blocks bom...

**Comment by jetersen** [comment]
@Dohbedoh would appreciate a review / release to unblock the BOM.

**Review by basil — APPROVED** [review]

**Comment by basil** [comment]
Now that BOM has been released I think we should close this in favor of #360.
